hack/heap/hh_shared.c

```diff
@@ -176,6 +176,10 @@
   }
 #endif
 
+#ifndef MAP_NORESERVE
+  #define MAP_NORESERVE 0
```

Comment:
Was wondering what this was doing, and found https://reviews.llvm.org/D6953 which explained that FreeBSD 11.x removes this flag, and it was a no-op before that. Makes sense. I'll add a comment.